lib/game.rb

```diff
@@ -0,0 +1,37 @@
+require './lib/frame.rb'
+class Game 
+    def initialize
+        @frames = []
+    end
+
+    def add(frame)
+        @frames << frame
+    end
+
+    def scoring_frames(number)
+        if @frames[number - 1].strike?
+            if @frames[number].strike?
+                20 + @frames[number + 1].roll_one
+            end
+        end
+
+         @frames[number].sum + 10
+
+        elsif @frames[number - 1].spare?
+            @frames[number].roll_one + 10
+        else
+            @frames[number - 1].sum
+        end
+    end
```

Comment:
You're nearly there with the logic by the looks of it. But I think having some RSpec tests will help you tell fairly quickly what's working and what's not.

It's also worth considering whether to keep track of the number of frames, so that you can add in the logic for if you get a strike or spare in the 10th frame.